From: Sage Weil <sage@redhat.com>
Date: Mon, 18 Apr 2016 09:16:02 -0400
Subject: udev: remove 60-ceph-partuuid-workaround-rules

These were added to get /dev/disk/by-partuuid/ symlinks to work on
wheezy.  They are no longer needed for the supported distros (el7+,
jessie+, trusty+), and they apparently break dm by opening devices they
should not.

Fixes: http://tracker.ceph.com/issues/15516
Signed-off-by: Sage Weil <sage@redhat.com>
(cherry picked from commit 9f77244b8e0782921663e52005b725cca58a8753)
(cherry picked from commit bf6bcaead158c1a64bd190e595249be623b29197)

RHCS cherry-pick does not include the /debian/* changes, since those
files are not present in upstream's v10.2.1 tarball.
---
 Makefile.am                            |  1 -
 ceph.spec.in                           |  2 --
 udev/60-ceph-partuuid-workaround.rules | 37 ----------------------------------
 3 files changed, 40 deletions(-)
 delete mode 100644 udev/60-ceph-partuuid-workaround.rules

diff --git a/Makefile.am b/Makefile.am
index 7734be2..81cb290 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -15,7 +15,6 @@ EXTRA_DIST += \
 	etc/sysconfig/SuSEfirewall2.d/services/ceph-mon \
 	etc/sysconfig/SuSEfirewall2.d/services/ceph-osd-mds \
 	udev/50-rbd.rules \
-	udev/60-ceph-partuuid-workaround.rules \
 	udev/95-ceph-osd.rules \
 	udev/95-ceph-osd-alt.rules \
 	share/known_hosts_drop.ceph.com \
diff --git a/ceph.spec.in b/ceph.spec.in
index ece232e..6ec065e 100644
--- a/ceph.spec.in
+++ b/ceph.spec.in
@@ -780,7 +780,6 @@ install -m 0644 -D etc/sysconfig/SuSEfirewall2.d/services/ceph-osd-mds %{buildro
 
 # udev rules
 install -m 0644 -D udev/50-rbd.rules $RPM_BUILD_ROOT%{_udevrulesdir}/50-rbd.rules
-install -m 0644 -D udev/60-ceph-partuuid-workaround.rules $RPM_BUILD_ROOT%{_udevrulesdir}/60-ceph-partuuid-workaround.rules
 
 %if (0%{?rhel} && 0%{?rhel} < 7)
 install -m 0644 -D udev/95-ceph-osd-alt.rules $RPM_BUILD_ROOT/lib/udev/rules.d/95-ceph-osd.rules
@@ -1257,7 +1256,6 @@ fi
 %{_sbindir}/ceph-disk
 %{_sbindir}/ceph-disk-udev
 %{_libexecdir}/ceph/ceph-osd-prestart.sh
-%{_udevrulesdir}/60-ceph-partuuid-workaround.rules
 %{_udevrulesdir}/95-ceph-osd.rules
 %{_mandir}/man8/ceph-clsinfo.8*
 %{_mandir}/man8/ceph-disk.8*
diff --git a/udev/60-ceph-partuuid-workaround.rules b/udev/60-ceph-partuuid-workaround.rules
deleted file mode 100644
index 2905969..0000000
--- a/udev/60-ceph-partuuid-workaround.rules
+++ /dev/null
@@ -1,37 +0,0 @@
-#
-# this is a kludge installed by ceph to fix the /dev/disk/by-partuuid 
-# symlinks on systems with old udev (< 180).  it's a stripped down 
-# version of a newer 60-persistent-storage.rules file that hopefully 
-# captures the same set of conditions for setting up those symlinks.
-#
-
-# forward scsi device event to corresponding block device
-ACTION=="change", SUBSYSTEM=="scsi", ENV{DEVTYPE}=="scsi_device", TEST=="block", ATTR{block/*/uevent}="change"
-
-ACTION=="remove", GOTO="persistent_storage_end_two"
-
-SUBSYSTEM!="block", GOTO="persistent_storage_end_two"
-
-# skip rules for inappropriate block devices
-KERNEL=="fd*|mtd*|nbd*|gnbd*|btibm*|md*", GOTO="persistent_storage_end_two"
-
-# ignore partitions that span the entire disk
-TEST=="whole_disk", GOTO="persistent_storage_end_two"
-
-# for partitions import parent information
-ENV{DEVTYPE}=="partition", IMPORT{parent}="ID_*"
-
-# skip unpartitioned removable media devices from drivers which do not send "change" events
-ENV{DEVTYPE}=="disk", KERNEL!="sd*|sr*", ATTR{removable}=="1", GOTO="persistent_storage_end_two"
-
-# probe filesystem metadata of disks
-KERNEL!="sr*", IMPORT{program}="/sbin/blkid -o udev -p $tempnode"
-
-# by-partlabel/by-partuuid links (partition metadata)
-ENV{ID_PART_ENTRY_SCHEME}=="gpt", ENV{ID_PART_ENTRY_UUID}=="?*", SYMLINK+="disk/by-partuuid/$env{ID_PART_ENTRY_UUID}"
-ENV{ID_PART_ENTRY_SCHEME}=="gpt", ENV{ID_PART_ENTRY_NAME}=="?*", SYMLINK+="disk/by-partlabel/$env{ID_PART_ENTRY_NAME}"
-
-# NEW: by-parttypeuuid links (type.id)
-ENV{ID_PART_ENTRY_SCHEME}=="gpt", ENV{ID_PART_ENTRY_TYPE}=="?*", ENV{ID_PART_ENTRY_UUID}=="?*", SYMLINK+="disk/by-parttypeuuid/$env{ID_PART_ENTRY_TYPE}.$env{ID_PART_ENTRY_UUID}"
-
-LABEL="persistent_storage_end_two"
